addons/van_dml/src/README.md

```diff
@@ -0,0 +1,116 @@
+    
+    bdiv() // Create a div and open for append
+        h1()
+    end()
+```
+begin(ID) can also digest Strings:
+```JS
+    <div id="ID"></div>
+    <script>
+    begin("ID") // select div by ID
+        h1()
+    end()
+    </script>
+```
+
+## end(n)
+
+`end()` finishes the append mode and returns to the previous base. For each `begin()` there should be one call of `end()`. You can verify this by checking `SP() === 0` (see below). 
+If no base is selected (or the calls `begin()` and `end()` are balanced), auto-Append is disabled. 
+
+end(n) runs `end()` n times. May be convenient to return from deep nested functions.
+
+## base() and sp()
+
+base() returns the current base:
+```JS
+    begin(div()) 
+    let b = base(); // returns the current base (div())
+```
+
+sp() is the current Stack-Position. Initially SP() is 0. With each call of `begin()` sp() is incremented, `end()` decements sp().
+To check your code, you can add this line to the end of Javascript:
+```JS
+    begin(document.body) 
+       .... some code here
+    end()
+    
+    if (sp() !== 0) alert(`baseStack mismatch: stacksize ${sp()} at end of Javascript`)
+```
+
+## css(): create CSS-rules dynamically with Javascript
+
+`css(s)` adds new CSS-rules during runtime.  
+s can be 
+- a string css(".myClass { color: red; }") 
+- a template string with multiple rules
+```JS
+    css(`
+        .myClass { color: red; }
+        .myClass: hover { color: green }
+    `)
+ ```
+ - It can contain template literals: 
+ ```JS
+    let color = [ "red", "green", "blue" ]
+    css(`
+        .myClass { color: ${mycolor[1]}; }
+    `)
+```
+If a rule already exists, it will be overwritten by the new definition. Results are immediately shown by the browser. Definitions with css() work like any other CSS you provide with static definitions, they just are added from within Javascript. This makes it easy to control CSS programmatically. css can be used anywhere and anytime in the script. New css-rules will become active immediately and might even be changed as shown in the example below:
+```JS
+    const { button } = van.tags
+    const { add, css } = van
+        add(document.body,
+        button({onclick: () => css(".class1 {background-color: red;}")},"set class1 red"),
+        button({onclick: () => css(".class1 {background-color: green;}")},"set class1 green")
+    );
+```
+**!Attention:**: css() adds new rules to the first stylesheet (`window.document.styleSheets[0]`). If the first stylesheet was loaded from an external source, this may cause CORS errors. For more information see [here](https://davidwalsh.name/add-rules-stylesheets) and [here](https://stackoverflow.com/questions/49088507/cannot-access-rules-in-external-cssstylesheet). Always add a ***local CSS-file*** first, as the dynamic rules are included in the first style sheet. If you need support, please use the [Discussions](https://github.com/vanjs-org/van/discussions) page.
```

Comment:
Support should be in its own section cuz we want to provide support for everything related DML, instead of just `css` function.

Reply:
Good point! Currently DML refers still to the old version, but we can have all van_dml-related things here. I will place a comment in the readme.md

Reply:
Is comment resolved in the latest version?

Reply:
Yes!

Reply:
What I meant is, the sentence "If you need support, please use the [Discussions](https://github.com/vanjs-org/van/discussions) page." should be in its own section, instead of part of the section for `css`. I haven't seen the change is reflected in the latest version. Do you want to make the change?

Reply:
I don't think this comment is resolved.